File: backend/src/MiniErp.Domain/Customers/Customer.cs
namespace MiniErp.Domain.Customers;

public class Customer
{
    public string Id { get; set; } = default!;
    public string CustomerCode { get; set; } = default!;
    public string CustomerName { get; set; } = default!;
    public string CompanyName { get; set; } = default!;

    public string Segment { get; set; } = default!;

    public string ContactPerson { get; set; } = default!;
    public string? ContactEmail { get; set; }
    public string? ContactPhone { get; set; }

    public string Region { get; set; } = default!;
    public string? Address { get; set; }

    public string Status { get; set; } = default!;
    public string? Notes { get; set; }

    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
}

File: frontend/src/types/customer.ts
export type CustomerStatus = "Active" | "Inactive" | "Prospect";
export type CustomerSegment = "Enterprise" | "SME" | "Startup";

export type Customer = {
    id: string;
    customerCode: string;
    customerName: string;
    companyName: string;
    segment: CustomerSegment;
    contactPerson: string;
    contactEmail?: string | null;
    contactPhone?: string | null;
    region: string;
    address?: string | null;
    status: CustomerStatus;
    notes?: string | null;
    createdAt: string;
    updatedAt: string;
};

export type CreateCustomerRequest = {
    customerCode: string;
    customerName: string;
    companyName: string;
    segment: CustomerSegment;
    contactPerson: string;
    contactEmail?: string | null;
    contactPhone?: string | null;
    region: string;
    address?: string | null;
    status: CustomerStatus;
    notes?: string | null;
};

export type UpdateCustomerRequest = CreateCustomerRequest;

export type CustomerFormValues = CreateCustomerRequest;